crates/pathfinder/src/p2p_network/sync_handlers/v1/conv.rs

```diff
@@ -0,0 +1,305 @@
+                .1
+                .l2_to_l1_messages
+                .into_iter()
+                .map(|m| MessageToL1 {
+                    from_address: m.from_address.0,
+                    payload: m.payload.into_iter().map(|p| p.0).collect(),
+                    to_address: EthereumAddress(m.to_address.0),
+                })
+                .collect(),
+            execution_resources: {
+                let e = self.1.execution_resources.unwrap_or_default();
+                // Assumption: the values are small enough to fit into u32
+                ExecutionResources {
+                    builtins: BuiltinCounter {
+                        bitwise: e
+                            .builtin_instance_counter
+                            .bitwise_builtin
+                            .try_into()
+                            .unwrap(),
+                        ecdsa: e.builtin_instance_counter.ecdsa_builtin.try_into().unwrap(),
+                        ec_op: e.builtin_instance_counter.ec_op_builtin.try_into().unwrap(),
+                        pedersen: e
+                            .builtin_instance_counter
+                            .pedersen_builtin
+                            .try_into()
+                            .unwrap(),
+                        range_check: e
+                            .builtin_instance_counter
+                            .range_check_builtin
+                            .try_into()
+                            .unwrap(),
+                        poseidon: e
+                            .builtin_instance_counter
+                            .poseidon_builtin
+                            .try_into()
+                            .unwrap(),
+                        keccak: e
+                            .builtin_instance_counter
+                            .keccak_builtin
+                            .try_into()
+                            .unwrap(),
+                        output: e
+                            .builtin_instance_counter
+                            .output_builtin
+                            .try_into()
+                            .unwrap(),
+                        segment_arena: e
+                            .builtin_instance_counter
+                            .segment_arena_builtin
+                            .try_into()
+                            .unwrap(),
+                    },
+                    steps: e.n_steps.try_into().unwrap(),
+                    memory_holes: e.n_memory_holes.try_into().unwrap(),
+                }
+            },
+            revert_reason: self.1.revert_error.unwrap_or_default(),
+            consumed_message: self.1.l1_to_l2_consumed_message.map(|x| MessageToL2 {
+                from_address: EthereumAddress(x.from_address.0),
+                payload: x.payload.into_iter().map(|p| p.0).collect(),
+                to_address: x.to_address.0,
+                entry_point_selector: x.selector.0,
+                // TODO option?
```

Comment:
```suggestion
```
nowadays the feeder gw always sets the value to 0, and if one happens to have some really old snapshot (I'm not sure that's possible) we should still default to 0